### PR TITLE
Take _embed into account in rest_preload_api_request()

### PR DESCRIPTION
**Pull request by lukaspawlik**
Trac ticket: https://core.trac.wordpress.org/ticket/51722

@TimothyBJacobs  please find attached implementation of your idea from mentioned Trac ticket.

Please let me know if you would like me to modify this in any different way.

cc @spacedmonkey

### CODE REVIEW

File: src/wp-includes/rest-api.php [hunk 1]
```diff
@@ -2519,12 +2519,10 @@ function rest_preload_api_request( $memo, $path ) {
...
 	$response = rest_do_request( $request );
 	if ( 200 === $response->status ) {
-		$server = rest_get_server();
-		$data   = (array) $response->get_data();
-		$links  = $server::get_compact_response_links( $response );
-		if ( ! empty( $links ) ) {
-			$data['_links'] = $links;
-		}
+		$server            = rest_get_server();
+		$embed_param_value = $request->get_param( '_embed' );
```
Comment (TimothyBJacobs) [inline src/wp-includes/rest-api.php:2523]:
```suggestion
		$embed_param_value = $request['_embed'];
```

We prefer not to use `get_param`.

**Comment by lukaspawlik** [comment]
@TimothyBJacobs Accepted your suggestion. This seems to be ready.

**Comment by TimothyBJacobs** [comment]
This looks great, thanks @lukaspawlik! Would you be interested in writing a test for this?

**Comment by TimothyBJacobs** [comment]
Merged in https://github.com/WordPress/wordpress-develop/commit/bb395706f4ebe6efeca2a951fad69ba4d98fb799. Thanks again for the patch @lukaspawlik!

**Pull request closed without merging**